src/client.ts

```diff
@@ -0,0 +1,157 @@
+import { ERROR_REASON, GENERAL_ERROR } from './constants';
+import { OpenFeature } from './open-feature';
+import {
+  Client,
+  Context,
+  EvaluationDetails,
+  FlagEvaluationOptions,
+  FlagValue,
+  Hook,
+  ResolutionDetails,
+  TransformingProvider,
+} from './types';
+
+type OpenFeatureClientOptions = {
+  name?: string;
+  version?: string;
+};
+
+export class OpenFeatureClient implements Client {
+  name?: string | undefined;
+  version?: string | undefined;
+  readonly context: Context;
+
+  constructor(private readonly api: OpenFeature, options: OpenFeatureClientOptions, context: Context = {}) {
+    this.name = options.name;
+    this.version = options.version;
+    this.context = context;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  addHooks(...hooks: Hook<FlagValue>[]): void {
+    throw new Error('Method not implemented.');
+  }
+
+  get hooks(): Hook<FlagValue>[] {
+    throw new Error('Method not implemented.');
+  }
+
+  async getBooleanValue(
+    flagKey: string,
+    defaultValue: boolean,
+    context?: Context,
+    options?: FlagEvaluationOptions
+  ): Promise<boolean> {
+    return (await this.getBooleanDetails(flagKey, defaultValue, context, options)).value;
+  }
+
+  getBooleanDetails(
+    flagKey: string,
+    defaultValue: boolean,
+    context?: Context,
+    options?: FlagEvaluationOptions
+  ): Promise<EvaluationDetails<boolean>> {
+    return this.evaluate<boolean>(flagKey, this.provider.resolveBooleanEvaluation, defaultValue, context, options);
```

Comment:
Instead of an ugly switch case, we can pass a reference to the provider function we want to run: `this.provider.resolveBooleanEvaluation` and then bind the provider to it when we call it: https://github.com/open-feature/node-sdk/pull/8/files#diff-25d66d74617fe2e23d7946bd6e3ba95640ab1b9bc8947445d604fc271c7c1f12R138